### PR TITLE
Add `--keep` flag to `crf-search` subcommand

### PR DESCRIPTION
**Pull request by t-nil**
For me it's very useful to look at the generated samples for manual verification. This command works on my machine right now.

@alexheretic please tell me if you want the flag refactored into `args::Sample`. Its probably better that way, right?

### CODE REVIEW

**Comment by t-nil** [comment]
Btw: what is the double purpose of cleaning in [main()](https://github.com/alexheretic/ab-av1/blob/2ad10eec75dea4719c025ed7949a7dfae90e7790/src/main.rs#L53) and [sample_encode::run](https://github.com/alexheretic/ab-av1/blob/2ad10eec75dea4719c025ed7949a7dfae90e7790/src/command/sample_encode.rs#L272)

**Comment by alexheretic** [comment]
Thanks, I 'm ok with crf-search & auto-encode having `--keep`. It does make we wonder if we should improve the output a bit though, e.g. informing which temp-dir is being kept so users don't have to guess / set one explicitly.

> please tell me if you want the flag refactored into args::Sample. Its probably better that way, right?

Yes I think that makes sense.

> Btw: what is the double purpose of cleaning in [main()](https://github.com/alexheretic/ab-av1/blob/2ad10eec75dea4719c025ed7949a7dfae90e7790/src/main.rs#L53) and [sample_encode::run](https://github.com/alexheretic/ab-av1/blob/2ad10eec75dea4719c025ed7949a7dfae90e7790/src/command/sample_encode.rs#L272)

Early cleaning keeps storage usage ceiling down. Cleaning at the end is required to handle ctrl-c interrupts & anything left over.

**Comment by t-nil** [comment]
Should be fine now :)

**Comment by alexheretic** [comment]
Looks good now, just one minor code comment.

Can you pop something in the changelog?

File: src/main.rs [hunk 1]
```diff
@@ -56,10 +56,16 @@ async fn main() -> anyhow::Result<()> {
...
     fn keep_temp_files(&self) -> bool {
         match self {
-            Self::SampleEncode(args) => args.keep,
-            _ => false,
+            Self::SampleEncode(args) => args.sample.keep,
+            Self::CrfSearch(args) => args.sample.keep,
+            Self::AutoEncode(args) => args.search.sample.keep,
+            Self::Vmaf(_) => false,
+            Self::Encode(_) => false,
+            Self::PrintCompletions(_) => false,
```
[the remark this answers is not in the document]
Comment (t-nil) [reply src/main.rs:68]: Good point. I thought hard about other possibilities to automatically detect usage of Args::Sample, but found none that had reasonable complexity.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Unreleased
 * Use a single ffmpeg process to calculate VMAF replacing multi process piping.
 * Exclude subtitle tracks from samples.
+* Add `--keep` option for _crf-search_.
```
Comment (alexheretic) [inline CHANGELOG.md:4]:
```suggestion
* Add `--keep` option for _crf-search_ & _auto-encode_.
```

**Review by alexheretic — APPROVED** [review]

**Comment by alexheretic** [comment]
Thanks!

**Pull request merged**